drivers/mongodb/mongodb.go

```diff
@@ -95,7 +95,8 @@ func (d *Mongodb) listFields(collection *mongo.Collection) ([]*schema.Column, er
 			return columns, err
 		}
 		total += 1
-		for key, value := range result.Map() {
+		for _, entry := range result {
+			key, value := entry.Key, entry.Value
```

Comment:
Note: I chose this option over unmarshalling as a `bson.M` map to maintain the order preservation.

**However** by using `result.Map()` the order was already undone, and before this change the key-value pairs were iterated over in hash table order.

If this is now resulting in unexpected ordering changes, we could change the above to use `var result bson.M` instead and use `for key, value := range result {` for this loop.